test/Test/Pos/BinarySpec.hs

```diff
@@ -29,22 +38,51 @@ spec = describe "Bi" $ do
             binaryTest @(B.FixedSizeInt Int)
             binaryTest @(B.FixedSizeInt Int64)
         describe "TinyVarInt" $ do
-            -- TODO: check that these actually take 1 byte
             describe "1 byte" $ do
+                it "serializing 0 takes 1 byte" $ serializeFail 0 1
                 prop "0" $ binaryEncodeDecode (B.TinyVarInt 0)
+                it "serializing 1 takes 1 byte" $ serializeFail 1 1
                 prop "1" $ binaryEncodeDecode (B.TinyVarInt 1)
+                it "serializing 0x75 takes 1 byte" $ serializeFail 0x75 1
                 prop "0x75" $ binaryEncodeDecode (B.TinyVarInt 0x75)
+                it "serializing 0x79 takes 1 byte" $ serializeFail 0x79 1
                 prop "0x79" $ binaryEncodeDecode (B.TinyVarInt 0x79)
-            -- TODO: check that these actually take 2 bytes
             describe "2 bytes" $ do
+                it "serializing 0x80 takes 2 bytes" $ serializeFail 0x80 2
                 prop "0x80" $ binaryEncodeDecode (B.TinyVarInt 0x80)
+                it "serializing 0x81 takes 2 bytes" $ serializeFail 0x81 2
                 prop "0x81" $ binaryEncodeDecode (B.TinyVarInt 0x81)
+                it "serializing 0x82 takes 2 bytes" $ serializeFail 0x82 2
                 prop "0x82" $ binaryEncodeDecode (B.TinyVarInt 0x82)
+                it "serializing 0x100 takes 2 bytes" $ serializeFail 0x100 2
                 prop "0x100" $ binaryEncodeDecode (B.TinyVarInt 0x100)
+                it "serializing 0x1000 takes 2 bytes" $ serializeFail 0x1000 2
                 prop "0x1000" $ binaryEncodeDecode (B.TinyVarInt 0x1000)
                 let maxnum = 2^(14::Int)-1
+                it "serializing (2^14)-1 takes 2 bytes" $
+                    B.biSize (B.TinyVarInt maxnum) `shouldBe` 2
                 prop "2^14-1" $ binaryEncodeDecode (B.TinyVarInt maxnum)
-            -- TODO: check that trying to serialize more than 2^14-1 fails
-            -- TODO: check that deserializing more than 2 bytes fails
-            -- TODO: check that deserializing an ambiguous 0 (i.e. @80 00@)
-            --       fails as well
+                prop "fails to serialize more than 2^14 - 1" $ do
+                    n <-  generate $ B.TinyVarInt <$> choose (maxnum + 1, maxBound)
+                    shouldThrowException B.encode anyErrorCall n
+            describe "deserialize more than 2 bytes" $ do
+                prop "fails to deserialize more than 2 bytes" $ do
+                    let wordGen =
+                            generate $
+                                (arbitrary :: Gen Word8) `suchThat` (flip testBit 7)
+                    w1 <- wordGen
+                    w2 <- wordGen
+                    let bs =
+                            uncurry BS.append . bimap BS.singleton BS.singleton $ (w1, w2)
+                    shouldThrowException (B.decode @B.TinyVarInt) anyErrorCall bs
```

Comment:
I did that, and the test failed because that's not the behavior set by the implementation. I left this in the code I submitted for review so that if someone (ie you) noticed it, I could be sure the `get` for `TinyVarInt` needed to be changed. I'll do it.